sympy/stats/drv.py

```diff
@@ -30,24 +31,29 @@ def check(*args):
     def sample(self):
         """ A random realization from the distribution """
         icdf = self._inverse_cdf_expression()
-        return floor(icdf(random.uniform(0, 1)))
+        while(True):
+            sample_ = floor(list(icdf(random.uniform(0, 1)))[0])
+            if sample_ >= self.set.inf:
+                return sample_
 
     @cacheit
     def _inverse_cdf_expression(self):
         """ Inverse of the CDF
 
         Used by sample
         """
-        x, z = symbols('x, z', real=True, positive=True, cls=Dummy)
+        x = symbols('x', positive = True,
```

Comment:
Python code standard (PEP8): "Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value." https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements